keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/titus/TitusClusterHandler.kt

```diff
@@ -283,49 +285,32 @@ class TitusClusterHandler(
       )
     }
 
-    val container = serverGroups.values.maxBy { it.capacity.desired ?: it.capacity.max }?.container
+    val container = serverGroups.values.maxByOrNull { it.capacity.desired ?: it.capacity.max }?.container
       ?: throw ExportError("Unable to locate container from the largest server group: $serverGroups")
 
     val registry = getRegistryForTitusAccount(exportable.account)
 
     val images = cloudDriverService.findDockerImages(
-      account = getRegistryForTitusAccount(exportable.account),
+      account = registry,
       repository = container.repository(),
       user = DEFAULT_SERVICE_ACCOUNT
     )
 
-    val matchingImages = images.filter { it.digest == container.digest }
-    if (matchingImages.isEmpty()) {
-      throw ExportError("Unable to find matching image (searching by digest) in registry ($registry) for $container")
-    }
-    val versionStrategy = guessVersioningStrategy(matchingImages)
-      ?: throw DockerArtifactExportError(matchingImages.map { it.tag }, container.toString())
-
-    return DockerArtifact(
-      name = container.repository(),
-      tagVersionStrategy = versionStrategy
-    )
-  }
-
-  /**
-   * Tries to find a matching versioning strategy from a list of docker tags that correspond to the same digest.
-   */
-  fun guessVersioningStrategy(images: List<DockerImage>): TagVersionStrategy? {
-    val versioningStrategies = mutableSetOf<TagVersionStrategy>()
+    val matchingImage = images.firstOrNull { it.digest == container.digest }
+      ?: throw ExportError("Unable to find matching image (searching by digest) in registry ($registry) for $container")
 
-    images.forEach { image ->
-      val versionStrategy = deriveVersioningStrategy(image.tag)
-      if (versionStrategy != null) {
-        versioningStrategies.add(versionStrategy)
-      }
-    }
-    return when (versioningStrategies.size) {
-      1 -> versioningStrategies.first()
-      0 -> null
-      else -> {
-        log.warn("Multiple versioning strategies apply for image, returning first")
-        versioningStrategies.first()
-      }
+    // prefer branch-based artifact spec, fallback to tag version strategy
+    return if (matchingImage.branch != null) {
```

Comment:
The `DockerImage` returned by CloudDriver already contained the branch name, so we're just checking for its presence here and using that to construct the artifact.